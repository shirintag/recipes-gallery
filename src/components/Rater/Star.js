import React from "react";

import "./StarRating.css";

const Star = ({ selected = false, onClick = (f) => f }) => (
  <div className="star-container" onClick={onClick}>
    <div className={selected ? "star selected" : "star"} />
  </div>
);

export default Star;
