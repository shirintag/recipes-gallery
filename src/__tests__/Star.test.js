import React from "react";
import { render } from "@testing-library/react";
import Star from "components/Rater/Star";

describe("Star", () => {
  it("should render", () => {
    const { container } = render(<Star />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should change class based on props", () => {
    var { container } = render(<Star selected={false} />);
    container = container.firstChild;
    expect(container.firstChild.className).toEqual("star");

    var { container } = render(<Star selected={true} />);
    container = container.firstChild;
    expect(container.firstChild.className).toEqual("star selected");
  });
});
