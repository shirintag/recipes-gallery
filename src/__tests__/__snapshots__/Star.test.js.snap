// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star should render 1`] = `
<div
  class="star-container"
>
  <div
    class="star"
  />
</div>
`;
